custom_components/hacs/repositories/base.py

```diff
@@ -796,8 +797,8 @@ async def remove_local_directory(self) -> None:
                     f"{self.hacs.configuration.theme_path}/"
                     f"{self.data.name}.yaml"
                 )
-                if os.path.exists(path):
-                    os.remove(path)
+                if await async_exists(self.hacs.hass, path):
+                    await async_remove(self.hacs.hass, path)
```

Comment:
It's bad to go into the executor twice. Can we add a feature to `async_remove` to not fail if a file doesn't exist ?

Reply:
If it's changed to use `Path(filepath).unlink(missing_ok=True)` it would be a single one.

Reply:
We could give `async_remove` a `missing_ok` flag